pkg/plugins/plugin_test.go

```diff
@@ -71,30 +72,23 @@ func TestGet(t *testing.T) {
 
 	t.Run("success", func(t *testing.T) {
 		plugin, err := Get(fruitPlugin, fruitImplements)
-		if err != nil {
-			t.Fatal(err)
-		}
-		if p.Name() != plugin.Name() {
-			t.Errorf("no matching plugin with name %s found", plugin.Name())
-		}
-		if plugin.Client() != nil {
-			t.Error("expected nil Client but found one")
-		}
-		if !plugin.IsV1() {
-			t.Error("Expected true for V1 plugin")
-		}
+		assert.NilError(t, err)
```

Comment:
I guess for these, we wouldn't be able to continue in any way, correct?

Otherwise, these could be `assert.Check(t, err)` to not fail early.

Reply:
If we get a non-nil error here, the plugin will be nil, so with `Check` we will have a nil dereference (at `plugin.Name()`).

Reply:
Ah, yes, I was thinking that was the case 👍